src/extension/csv-with-metadata-header/src/main/java/au/org/emii/geoserver/wfs/response/CSVWithMetadataHeaderOutputFormat.java

```diff
@@ -70,7 +75,89 @@ protected void write(FeatureCollectionResponse featureCollection,
+    }
+
+
+    public void writeData(FeatureCollectionResponse featureCollection, OutputStream output, Operation getFeature) throws IOException, ServiceException {
+        BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output, this.gs.getGlobal().getSettings().getCharset()));
+        FeatureCollection<?, ?> fc = (FeatureCollection)featureCollection.getFeature().get(0);
+        Object att;
+
+        SimpleFeatureType ft = (SimpleFeatureType)((FeatureCollection)fc).getSchema();
+        w.write("FID,");
+
+        for(int i = 0; i < ft.getAttributeCount(); ++i) {
+            AttributeDescriptor ad = ft.getDescriptor(i);
+            w.write(this.prepCSVField(ad.getLocalName()));
+            if (i < ft.getAttributeCount() - 1) {
+                w.write(",");
+            }
+        }
+
+        w.write("\r\n");
+        NumberFormat coordFormatter = NumberFormat.getInstance(Locale.US);
+        coordFormatter.setMaximumFractionDigits(this.getInfo().getGeoServer().getSettings().getNumDecimals());
+        coordFormatter.setGroupingUsed(false);
+        FeatureIterator i = ((FeatureCollection)fc).features();
+
+        try {
+            while (i.hasNext()) {
+                Feature f = i.next();
+                w.write(this.prepCSVField(f.getIdentifier().getID()));
+                w.write(",");
+
+                for(int j = 0; j < ((SimpleFeature)f).getAttributeCount(); ++j) {
+                    att = ((SimpleFeature)f).getAttribute(j);
+                    if (att != null) {
+                        String value = formatToString(att, coordFormatter);
+                        w.write(this.prepCSVField(value));
+                    }
+
+                    if (j < ((SimpleFeature)f).getAttributeCount() - 1) {
+                        w.write(",");
+                    }
+                }
+                w.write("\r\n");
+            }
+        } finally {
+            i.close();
+        }
+        w.flush();
+    }
+
+    private String prepCSVField(String field) {
+        String mod = field.replaceAll("\"", "\"\"");
+        if (CSV_ESCAPES.matcher(mod).find()) {
+            mod = "\"" + mod + "\"";
+        }
+        return mod;
+    }
+
+    public String formatToString(Object att, NumberFormat coordFormatter) {
+        String value = null;
+
+        if (att instanceof Number) {
+            value = coordFormatter.format(att);
+        } else if (att instanceof Date) {
+            if (att instanceof java.sql.Date) {
+                value = DateUtil.serializeSqlDate((java.sql.Date)att);
+            } else if (att instanceof Time) {
+                value = DateUtil.serializeSqlTime((Time)att);
+            }else {
+                value = utcDateString((Date)att);
+            }
+        } else {
+            value = att.toString();
+        }
+        return value;
+    }
+
+    private String utcDateString(Date date) {
+        DateTime dt = new DateTime( date, DateTimeZone.UTC );
```

Comment:
Actually, probably need to do a null check here (returning an empty string).  Apparently, new dateTime(null) returns the current date/time.   See https://stackoverflow.com/questions/5042587/convert-from-java-util-date-to-jodatime